tests/Feature/Engine/ExclusiveGatewayTest.php

```diff
@@ -0,0 +1,166 @@
+        $process = $this->createProcessWithExclusiveGateway();
+        $this->engine->createExecutionInstance($process, $dataStore);
+
+        //Get References
+        $start = $process->getEvents()->item(0);
+        $activityA = $process->getActivities()->item(0);
+
+        //Start the process
+        $start->start();
+
+        $this->engine->runToNextState();
+
+        //Assertion: Verify the triggered engine events. Two activities are activated.
+        $this->assertEvents([
+            EventNodeInterface::EVENT_EVENT_TRIGGERED,
+            GatewayInterface::EVENT_GATEWAY_TOKEN_ARRIVES,
+            GatewayInterface::EVENT_GATEWAY_ACTIVATED,
+            GatewayInterface::EVENT_GATEWAY_TOKEN_CONSUMED,
+            GatewayInterface::EVENT_GATEWAY_TOKEN_PASSED,
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+        ]);
+
+        //Completes the Activity A
+        $tokenA = $activityA->getTokens($dataStore)->item(0);
+        $activityA->complete($tokenA);
+
+        //the run to next state should go false when the max steps is reached.
+        $this->assertFalse($this->engine->runToNextState(1));
+
+        $this->engine->runToNextState();
+
+//        //Assertion: Verify the triggered engine events. The activity is closed and process is ended.
+        $this->assertEvents([
+            ActivityInterface::EVENT_ACTIVITY_COMPLETED,
+            ActivityInterface::EVENT_ACTIVITY_CLOSED,
+            EventNodeInterface::EVENT_EVENT_TRIGGERED
+        ]);
+    }
+
+    /**
+     * Tests the exclusive gateway triggering the default transition
+     */
+    public function testExclusiveGatewayWithDefaultTransition()
+    {
+        //Create a data store with data.
+        $dataStore = $this->dataStoreRepository->createDataStoreInstance();
+        $dataStore->putData('A', '2');
+        $dataStore->putData('B', '2');
+
+        //Load the process
+        $process = $this->createProcessWithExclusiveGateway();
+        $this->engine->createExecutionInstance($process, $dataStore);
+
+        //Get References
+        $start = $process->getEvents()->item(0);
+        $activityC = $process->getActivities()->item(2);
+
+        //Start the process
+        $start->start();
+
+        $this->engine->runToNextState();
+
+        //Assertion: Verify the triggered engine events. Two activities are activated.
+        $this->assertEvents([
+            EventNodeInterface::EVENT_EVENT_TRIGGERED,
+            GatewayInterface::EVENT_GATEWAY_TOKEN_ARRIVES,
+            GatewayInterface::EVENT_GATEWAY_ACTIVATED,
+            GatewayInterface::EVENT_GATEWAY_TOKEN_CONSUMED,
+            GatewayInterface::EVENT_GATEWAY_TOKEN_PASSED,
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+        ]);
+
+        //Completes the Activity A
+        $tokenC = $activityC->getTokens($dataStore)->item(0);
+        $activityC->complete($tokenC);
+
+        //the run to next state should go false when the max steps is reached.
```

Comment:
Since the flow does not have a loop, why do you check this assertion?

Reply:
corrected